### ISSUE TITLE
Wire Baseline into Make/Compose; Docs & Changelog; Permissions & Headless Plotting (Modifications-only)

### ISSUE THREAD

**Issue by jbrdge** [jbrdge/churn-prediction#42]
**Type:** Integration / DX improvements  
**Depends on:** "Introduce Baseline Modeling CLI and Supporting Scripts (Additions-only)"  
**Scope:** Modify tracked files only (no new code under tracked paths except minor `.gitkeep`).

---

## Summary
Connect the baseline CLI and scripts into the Docker-first workflow, improve DX, and update docs:

- **Makefile**: add targets `make-features-from-archive`, `train-baseline`, `train-baseline-sample`, `show-metrics`, `ls-artifacts`, `monte-carlo`, `monte-carlo-summary`, `mc-best`, `mc-show-best`, `mc-show-best-metrics`, `mc-ls-best`. Avoid heredocs; use `--entrypoint sh -c`.
- **docker-compose.yml**: optionally run as host user to avoid root-owned artifacts (`user: "${HOST_UID:-1000}:${HOST_GID:-1000}"`); set headless plotting env (`MPLBACKEND=Agg`, `MPLCONFIGDIR=/tmp/matplotlib`, `XDG_CACHE_HOME=/tmp`).
- **requirements.txt**: pin core libs for reproducibility.
- **README.md**: add Baseline Model section (quickstart, artifacts, sampling, Monte Carlo, best-run helpers, permissions note).
- **CHANGELOG.md**: add `[0.5.0]` entry with concise bullets.
- **.env.example**: include `HOST_UID/HOST_GID` defaults (optional note).
- **src/db.py**: clarify/leave as placeholder or remove if appropriate (team decision).

---

## Acceptance Criteria
- `make make-features-from-archive [N=…]` produces `data/processed/features.parquet` from Kaggle CSV.
- `make train-baseline` finishes and writes all artifacts under `artifacts/baseline_v1/`.
- `make show-metrics` pretty-prints metrics; `make ls-artifacts` lists artifacts.
- `make monte-carlo N=50 MC_ITERS=3` produces `artifacts/mc_baseline/metrics.csv` and per-run artifacts.
- `make mc-best [METRIC=f1|roc_auc]` copies best run to `artifacts/mc_baseline/best/` with `best_summary.json`.
- Plots render (no blank images) in headless containers.
- Files written by containers are owned by the host user on Linux/WSL when `HOST_UID/GID` is set.
- README/CHANGELOG updated accordingly.

---

## Tasks
- [ ] Update Makefile with all targets (no heredocs; container `--entrypoint sh -c`)
- [ ] Update docker-compose.yml with optional `user:` and plotting env vars
- [ ] Add `scripts/docker-env.sh` and reference it in README
- [ ] Pin versions in requirements.txt (pandas, scikit-learn, matplotlib, pyarrow, joblib)
- [ ] Add README Baseline section + Monte Carlo + mc-best helpers + permissions
- [ ] Update CHANGELOG `[0.5.0]`
- [ ] (Optional) Add `.gitkeep` under `artifacts/` and `data/processed/`
- [ ] Decide on `src/db.py` (placeholder comment or remove)

---

## How to Test
```bash
# Linux/WSL only
source scripts/docker-env.sh

# Full path
make make-features-from-archive N=500
make train-baseline
make show-metrics
make monte-carlo N=50 MC_ITERS=3
make mc-best METRIC=roc_auc
make mc-show-best
```

---

## Out of Scope
- New modeling algorithms / hyperparameter tuning
- DB schema or ETL logic changes
- CI pipeline (can be follow-up)

---

## Risks & Mitigations
- **Permissions variance across dev machines:**  
  Use `${HOST_UID}:${HOST_GID}` mapping and document in README. Provide `scripts/docker-env.sh` helper.
- **Matplotlib headless quirks:**  
  Set `MPLBACKEND=Agg` and writable cache dirs via env; use explicit figure creation in CLI.
- **Makefile fragility:**  
  Avoid heredocs; keep commands on single lines; rely on small scripts in `scripts/`.
